### PR TITLE
Move to a DI approach for resin-request and resin-token

### PR DESCRIPTION
**Pull request by pimterry**
This moves Resin-Pine to DI, as discussed in https://github.com/resin-io-modules/resin-request/issues/85

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,13 +39,18 @@
     "mocha": "^3.0.0",
     "mochainon": "^1.0.0",
     "resin-config-karma": "^1.0.4",
-    "resin-settings-client": "^3.5.2"
+    "resin-request": "^7.0.0",
+    "resin-settings-client": "^3.5.2",
```
Comment (emirotin) [inline package.json]: Is it still needed?

File: tests/pine.spec.coffee [hunk 2]
```diff
@@ -27,12 +26,13 @@ fetchMock.fetchMock.Promise = Promise
 require('resin-request/build/utils').fetch = fetchMock.fetchMock # Can become just fetchMock after issue above is fixed.
 
 token = getToken({ dataDirectory })
+request = require('resin-request')({ dataDirectory })
```
Comment (emirotin) [inline tests/pine.spec.coffee]: NIT but let's make these two lines consistent?

File: package.json (hunk 1 above)
Comment (pimterry) [reply package.json]: Good spot! No, removed.

File: tests/pine.spec.coffee (hunk 2 above)
Comment (pimterry) [reply tests/pine.spec.coffee]: Fixed

File: tests/pine.spec.coffee [hunk 3]
```diff
@@ -16,23 +15,23 @@ if IS_BROWSER
...
 
 fetchMock = require('fetch-mock').sandbox(Promise)
 # Promise sandbox config needs a little help. See:
 # https://github.com/wheresrhys/fetch-mock/issues/159#issuecomment-268249788
 fetchMock.fetchMock.Promise = Promise
 require('resin-request/build/utils').fetch = fetchMock.fetchMock # Can become just fetchMock after issue above is fixed.
 
-token = getToken({ dataDirectory })
+token = require('resin-token')({ dataDirectory })
+request = require('resin-request')({ dataDirectory })
```
Comment (emirotin) [inline tests/pine.spec.coffee]: As we already have the resin-request conversion to DI ready I'd bring it here and merge after that, would also serve as an integration test. Other than that I have no objections so approving this PR now.

**Review by emirotin — APPROVED** [review]

**Pull request merged**